posix: relocate `uname` cmd from sample

ycsin opened the pull request:
Relocate the `uname` cmd from the uname sample and updated the sample to use only POSIX headers & APIs. The sample can now be compiled on Linux:

```
# Go to the sample dir
cd ${ZEPHYR_BASE}/samples/posix/uname

# Compile the sample
make -f Makefile.host

# Run the binary
./build/uname

sysname[65]: Linux
nodename[65]: LAPTOP-YC
release[65]: 5.10.16.3-microsoft-standard-WSL2
version[65]: https://github.com/ycsin/zephyr/pull/1 SMP Fri Apr 2 22:23:49 UTC 2021
machine[65]: x86_64
```

![image](https://github.com/zephyrproject-rtos/zephyr/assets/1563974/d8334514-4528-4b58-b8b9-7be66abfeeda)

ycsin committed:
samples: posix: uname: convert to use `printf`

Convert to `printf` from `printk` to be more POSIX-y.

Signed-off-by: Yong Cong Sin <ycsin@meta.com>

ycsin committed:
posix: uname: move `uname` shell from sample

Relocate the `uname` shell implementation from uname sample, so
that it can be reused by other application and the uname sample
only uses POSIX APIs.

Signed-off-by: Yong Cong Sin <ycsin@meta.com>

ycsin committed:
posix: uname: shell: update help message with more information

Populate the help message with more information when an error
occurs.

Signed-off-by: Yong Cong Sin <ycsin@meta.com>

ycsin committed:
samples: posix: uname: add Makefile to compile on POSIX OS

Previously the sample was using some headers that aren't
available to the host, now we can add a `Makefile.host` to
compile the example on a POSIX OS like Linux:

```
# Go to the sample dir
cd ${ZEPHYR_BASE}/samples/posix/uname

# Compile the sample
make -f Makefile.host

# Run the binary
./build/uname

sysname[65]: Linux
nodename[65]: LAPTOP-YC
release[65]: 5.10.16.3-microsoft-standard-WSL2
version[65]: #1 SMP Fri Apr 2 22:23:49 UTC 2021
machine[65]: x86_64
```

Signed-off-by: Yong Cong Sin <ycsin@meta.com>

ycsin commented on lib/posix/uname.c:
```diff
@@ -51,3 +51,156 @@ int uname(struct utsname *name)
 	return 0;
 }
+
+#ifdef CONFIG_POSIX_UNAME_SHELL
```
Good idea, thanks! Moved it to `posix/shell`

ycsin commented:
Inspired by #67461 and added a commit to make the sample compilable on Linux with `make`

jukkar commented on lib/posix/shell/uname.c:
```diff
@@ -0,0 +1,157 @@
+	return 0;
+}
+
+SHELL_CMD_REGISTER(uname, NULL, NULL, uname_cmd_handler);
```
Just pondering here that should we have `posix` command handler and the `uname` would be a sub command for it.
If we do it like in this PR, we would have quite many commands in the "main" shell which imho clutters the shell quite a bit.

ycsin replied:
I've pushed another commit for this, and chose `posix` over `sh` as the former seems to be more intuitive

ycsin committed:
posix: shell: introduce top level `posix` command

Added a top level `posix` shell command for other POSIX
commands. Currently only `uname` is supported.

New POSIX commands can be added by including the
`posix_shell.h` header and use the `POSIX_CMD_ADD` helper
macro.

Signed-off-by: Yong Cong Sin <ycsin@meta.com>

jukkar reviewed (dismissed):
LGTM

ycsin commented:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/1563974/d8334514-4528-4b58-b8b9-7be66abfeeda)

ycsin commented on lib/posix/shell/Kconfig.uname:
```diff
@@ -0,0 +1,13 @@
+config POSIX_UNAME_SHELL
+	bool "Support for `uname` command"
+	select SHELL_GETOPT
+	select POSIX_SHELL
```
force pushed to make it such that the top level `posix` command is compiled only if there's a sub command
